internal/build/lifecycle_execution_test.go

```diff
@@ -800,20 +800,95 @@ func testLifecycleExecution(t *testing.T, when spec.G, it spec.S) {
 		})
 
 		when("-previous-image is used", func() {
-			it("passes previous-image to creator", func() {
-				lifecycle := newTestLifecycleExec(t, true, func(options *build.LifecycleOptions) {
-					options.PreviousImage = "previous-image"
+			when("previous-image is invalid", func() {
+				it("errors", func() {
+					var imageName name.Tag
+					imageName, err := name.NewTag("/some/image", name.WeakValidation)
+					h.AssertNil(t, err)
+					lifecycle := newTestLifecycleExec(t, true, func(options *build.LifecycleOptions) {
+						options.PreviousImage = "%%%"
+						options.Image = imageName
+					})
+					fakePhaseFactory := fakes.NewFakePhaseFactory()
+					err = lifecycle.Create(context.Background(), false, "", false, "test", "test", "test", fakeBuildCache, fakeLaunchCache, []string{}, []string{}, fakePhaseFactory)
+					h.AssertError(t, err, fmt.Sprintf("%s", err))
 				})
-				fakePhaseFactory := fakes.NewFakePhaseFactory()
-				err := lifecycle.Create(context.Background(), false, "", false, "test", "test", "test", fakeBuildCache, fakeLaunchCache, []string{}, []string{}, fakePhaseFactory)
-				h.AssertNil(t, err)
+			})
 
-				lastCallIndex := len(fakePhaseFactory.NewCalledWithProvider) - 1
-				h.AssertNotEq(t, lastCallIndex, -1)
+			// when("image is invalid", func() {
```

Comment:
Tried to add a unit test to check for invalid images, but couldn't figure out how to pass an invalid image name in the test as it gets validated. Tried passing tags like "%%%", tags with other invalid characters, tags with multiple slashes, etc.

Reply:
Once again, I'd rather not bother with a test for this 😅 

A test to check that we're bubbling up the error would be of _low value_, imo. Such a test would be more valuable if it were _our logic_ deciding what a valid/invalid image tag is.

In this case, I'd be comfortable having faith in our third-party library having its own tests. And then saving ourselves the keystrokes for every error check.

Reply:
Cool, will remove.